rust/lance/src/index.rs

```diff
@@ -252,9 +257,26 @@ impl DatasetIndexExt for Dataset {
         Ok(())
     }
 
-    async fn load_indices(&self) -> Result<Vec<IndexMetadata>> {
-        let manifest_file = self.manifest_file(self.version().version).await?;
-        read_manifest_indexes(&self.object_store, &manifest_file, &self.manifest).await
+    async fn load_indices<'a>(&'a self) -> Result<Vec<IndexMetadata>> {
+        let mut indices = self.index_metadata_cache.lock().await;
```

Comment:
Async locks make me a little uneasy.  Could we avoid it by doing something like...

```
Grab lock
If available and correct version then return
Release lock
Load indices from file
Grab lock
If not set, then set
Release lock
```

This could lead to multiple threads loading the indices but that should be rare and harmless.

Reply:
sg.